### DISCUSSION TITLE
dynamic popup height

### DISCUSSION THREAD

**Discussion by Silther** [Clooos/Bubble-Card#649] [Feature requests]
Currently there is a lot of space lost because you have to hardcode the height, would be cool if it could dynamically adjust the height to the selected cards.

> ![image](https://github.com/user-attachments/assets/ab7e07d2-55a4-4c71-912b-8c078f04cf24)

I mean I have some conditional cards inside the popup, so it should be smaller if the some cards are hidden and larger if not.

> ![image](https://github.com/user-attachments/assets/c17769c7-7445-4749-9bac-8dc41831f48e)

Also a hardcoded height doesn't handle zoom inside the browser very well.

> ![image](https://github.com/user-attachments/assets/ab5448af-1c58-4112-819a-3c6bf48cd962)

**Comment by Silther** [comment]
type: vertical-stack
cards:
  - type: custom:bubble-card
    card_type: pop-up
    hash: '#music'
    name: Music Zone
    icon: mdi:speaker
    show_icon: false
    show_name: false
    scrolling_effect: false
    card_layout: normal
    sub_button: []
    hide_backdrop: false
    show_state: false
    show_last_changed: false
    show_attribute: false
    force_icon: false
    tap_action:
      service: da
      target:
        entity_id: media_player.music_zone
      action: none
    double_tap_action:
      action: none
    hold_action:
      action: none
    button_type: name
    bg_opacity: '75'
    bg_blur: '25'
    shadow_opacity: '25'
    margin_top_desktop: 50px
    margin_top_mobile: 50px
    rows: 4
  - type: horizontal-stack
    cards:
      - type: custom:mushroom-title-card
        title: ''
        subtitle: Music
        title_tap_action:
          action: navigate
          navigation_path: /mobile-dashboard/office-music
      - type: custom:mushroom-title-card
        title: ''
        subtitle: ''
      - type: custom:mushroom-title-card
        title: ''
        subtitle: ''
      - type: custom:mushroom-title-card
        title: ''
        subtitle: ''
      - type: custom:mushroom-media-player-card
        entity: media_player.music_zone
        fill_container: false
        volume_controls:
          - volume_mute
        icon_type: none
        primary_info: none
        collapsible_controls: true
        use_media_info: false
        show_volume_level: false
        secondary_info: none
        hold_action:
          action: toggle
        tap_action:
          action: toggle
        double_tap_action:
          action: toggle
        card_mod:
          style: |
            ha-card {
              background: rgba(0,0,0,0);
              border: none;
             }
  - type: custom:vertical-stack-in-card
    cards:
      - type: custom:mushroom-media-player-card
        entity: media_player.music_zone
        use_media_info: true
        volume_controls:
          - volume_set
        collapsible_controls: true
        fill_container: true
        show_volume_level: true
        icon_type: entity-picture
        layout: horizontal
        name: Music Zone
      - type: custom:mushroom-media-player-card
        entity: media_player.music_zone
        fill_container: true
        media_controls:
          - previous
          - play_pause_stop
          - next
          - repeat
          - shuffle
          - on_off
        show_volume_level: false
        use_media_info: false
        collapsible_controls: false
        icon_type: none
        primary_info: none
        secondary_info: none
  - type: custom:mushroom-title-card
    title: ''
  - type: conditional
    conditions:
      - entity: media_player.music_zone
        state_not: 'off'
      - entity: media_player.music_zone
        state_not: unavailable
      - entity: media_player.music_zone
        state_not: standby
    card:
      type: vertical-stack
      cards:
        - type: custom:vertical-stack-in-card
          cards:
            - type: custom:mushroom-media-player-card
              entity: media_player.nest_mini_2
              show_volume_level: false
              volume_controls: []
              fill_container: true
              name: Nest Mini 2
            - type: custom:mushroom-media-player-card
              entity: media_player.nest_mini_2
              show_volume_level: true
              volume_controls:
                - volume_set
                - volume_mute
              fill_container: true
              name: Nest Mini 2
              primary_info: none
              secondary_info: none
              icon_type: none
        - type: custom:vertical-stack-in-card
          cards:
            - type: custom:mushroom-media-player-card
              entity: media_player.bose_home_speaker
              show_volume_level: false
              volume_controls: []
              fill_container: true
              name: Bose Home
            - type: custom:mushroom-media-player-card
              entity: media_player.bose_home_speaker
              show_volume_level: true
              volume_controls:
                - volume_set
                - volume_mute
              fill_container: true
              name: Bose Home
              primary_info: none
              secondary_info: none
              icon_type: none
        - type: conditional
          conditions:
            - condition: state
              entity: media_player.monitor
              state_not: unavailable
          card:
            type: custom:vertical-stack-in-card
            cards:
              - type: custom:mushroom-media-player-card
                entity: media_player.monitor
                show_volume_level: false
                volume_controls: []
                fill_container: true
                name: Monitor
              - type: custom:mushroom-media-player-card
                entity: media_player.monitor
                show_volume_level: true
                volume_controls:
                  - volume_set
                  - volume_mute
                fill_container: true
                name: Monitor
                primary_info: none
                secondary_info: none
                icon_type: none
  - type: conditional
    conditions:
      - entity: media_player.music_zone
        state_not: playing
      - entity: media_player.music_zone
        state_not: paused
      - entity: media_player.music_zone
        state_not: buffering
      - entity: media_player.music_zone
        state_not: idle
    card:
      type: vertical-stack
      cards:
        - type: custom:vertical-stack-in-card
          cards:
            - type: custom:mushroom-media-player-card
              entity: media_player.nest_mini_2
              use_media_info: false
              volume_controls:
                - volume_set
                - volume_mute
                - volume_buttons
              collapsible_controls: false
              fill_container: true
              show_volume_level: false
              icon_type: entity-picture
              layout: horizontal
              name: Nest Mini 2
            - type: custom:mushroom-media-player-card
              entity: media_player.nest_mini_2
              fill_container: true
              media_controls:
                - previous
                - play_pause_stop
                - next
                - repeat
                - shuffle
                - on_off
              show_volume_level: false
              use_media_info: true
              collapsible_controls: false
              icon_type: none
              primary_info: none
              secondary_info: none
        - type: custom:vertical-stack-in-card
          cards:
            - type: custom:mushroom-media-player-card
              entity: media_player.bose_home_speaker
              use_media_info: false
              volume_controls:
                - volume_set
                - volume_mute
                - volume_buttons
              collapsible_controls: false
              fill_container: true
              show_volume_level: false
              icon_type: entity-picture
              layout: horizontal
              name: Bose Home
            - type: custom:mushroom-media-player-card
              entity: media_player.bose_home_speaker
              fill_container: true
              media_controls:
                - previous
                - play_pause_stop
                - next
                - repeat
                - shuffle
                - on_off
              show_volume_level: false
              use_media_info: true
              collapsible_controls: false
              icon_type: none
              primary_info: none
              secondary_info: none
        - type: conditional
          conditions:
            - condition: state
              entity: media_player.monitor
              state_not: unavailable
          card:
            type: custom:vertical-stack-in-card
            cards:
              - type: custom:mushroom-media-player-card
                entity: media_player.monitor
                use_media_info: false
                volume_controls:
                  - volume_set
                  - volume_mute
                  - volume_buttons
                collapsible_controls: false
                fill_container: true
                show_volume_level: false
                icon_type: entity-picture
                layout: horizontal
                name: Monitor
              - type: custom:mushroom-media-player-card
                entity: media_player.monitor
                fill_container: true
                media_controls:
                  - previous
                  - play_pause_stop
                  - next
                  - repeat
                  - shuffle
                  - on_off
                show_volume_level: false
                use_media_info: true
                collapsible_controls: false
                icon_type: none
                primary_info: none
                secondary_info: none

**Comment by Disincentivato** [comment]
Hi, I was looking for exactly the same thing, I have the battery operated camera popup which would be nice to see adjusted in height too.
<img width="1045" height="829" alt="immagine" src="https://github.com/user-attachments/assets/1bb5cf40-9686-45b1-bad9-e491f15f8b0c" />

**Comment by bdunn44** [comment]
Quick PSA: I wanted to do this as well and found you can with a few simple CSS properties:

```
.bubble-pop-up {
  height: inherit !important;
  margin-bottom: 30px; # optional - adds a little margin at the bottom if desired
}
```

You need to add this to the pop up's styles as shown here:

<img width="1009" height="996" alt="image" src="https://github.com/user-attachments/assets/c0dbf1e8-2de7-4d1e-a5f8-585b5100addc" />

**Reply by userdema** [reply]
> Quick PSA: I wanted to do this as well and found you can with a few simple CSS properties:
> 
> ```
> .bubble-pop-up {
>   height: inherit !important;
>   margin-bottom: 30px; # optional - adds a little margin at the bottom if desired
> }
>```

Wow that's solved my problems.. thank you!!
The only modification that I made is apply a negative margin to avoid unnecessary space on bottom, and I applied it on

```
 .bubble-pop-up:not(.editor) 
```

To not have editor glitches...
The complete solution:

```
.bubble-pop-up:not(.editor) {
  height: inherit !important;
  margin-bottom: -70px;
  }
```

Thanks again!!

**Comment by zyi840449** [comment]
Use the following settings:

```
styles: |
.bubble-pop-up {
--custom-height-offset-mobile: ${hass.states["sensor.weather_report_height"].state} !important;
--custom-height-offset-desktop: calc(${hass.states["sensor.weather_report_height"].state} - 10vh) !important;
}
```

This makes the margin property of your pop-up card "variable," taking the value of the `sensor.weather_report_height` entity (which is currently 46vh).

The exact value of this sensor value is up to you to design your own formula.
You can determine the height based on the state of certain entities.
I determine the height based on the length of the mark-down text.

**Comment by Disincentivato** [comment]
I found another solution using the Pop-Up Box module, in this case you can also quickly change other parameters.
I didn't find a direct setting for the height, but I noticed that it adjusts it automatically. You can set the popup width.

<img width="974" height="459" alt="immagine" src="https://github.com/user-attachments/assets/04eab642-038d-476e-b172-aa82d1f68cfe" />

<img width="916" height="576" alt="immagine" src="https://github.com/user-attachments/assets/3fac6c3a-4d0b-46c8-8fed-07fb08962b94" />

**Comment by Clooos** [comment]
Hi! Just to say that this feature will (finally) be part of v3.2.0! 😄
